File: lib/src/main/java/com/github/goto1134/structurizr/export/d2/model/D2Connection.java
package com.github.goto1134.structurizr.export.d2.model;

/**
 * Representd different types of D2 relationship connections
 *
 * @see <a href="https://d2lang.com/tour/connections">D2 Connections</a>
 */
public enum D2Connection {
    WITHOUT_DIRECTION("--"),
    DIRECT("->"),
    REVERSE("<-"),
    BIDIRECTIONAL("<->>");
    private final String value;

    D2Connection(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: lib/src/main/kotlin/io/github/goto1134/structurizr/export/d2/model/D2Connection.kt
package io.github.goto1134.structurizr.export.d2.model

/**
 * Represents different types of D2 relationship connections
 *
 * @see [D2 Connections](https://d2lang.com/tour/connections/)
 */
enum class D2Connection(private val value: String) {
    WITHOUT_DIRECTION("--"),
    DIRECT("->"),
    REVERSE("<-"),
    BIDIRECTIONAL("<->");

    override fun toString() = value
}
